src/init/autoApplyCustomizations.ts

```diff
@@ -6,7 +6,7 @@ export function manageAutoApplyCustomizations(
   userConfig: IVSIcons,
   applyCustomizationCommand: () => void): void {
   if (!isNewVersion) { return; }
-  const propObj = packageJson.contributes.configuration.properties || {};
+  const propObj: any = packageJson.contributes.configuration.properties as object;
```

Comment:
As it is now you are totally right, and indeed it makes no sense to anyone that sees it.
Let me explain further my intention.

The code should be 
```
const propObj = packageJson.contributes.configuration.properties as object
for (const key in propObj) {
```
which is totally valid.

If I use
```
const propObj: Object= packageJson.contributes.configuration.properties as object
for (const key in propObj) {
```
then `tslint` `ban-types` rule complains.

Not here that we want to explicitly annotate `propObj` as an `Object` and not as `any`.

But because `ts` is complaining* I'm forced to change it to 
```
const propObj: any = packageJson.contributes.configuration.properties as object
for (const key in propObj) {
```
to workaround the issue.

as soon as `ts` fixes the issue `: any` can be removed.

*https://github.com/Microsoft/TypeScript/issues/14792